src/sage/algebras/quatalg/quaternion_algebra.py

```diff
@@ -1067,86 +1055,312 @@ def inner_product_matrix(self):
+            True
+
+            sage: L = QuadraticField(-15)
+            sage: QuaternionAlgebra(L, -1, -1).is_totally_definite()
+            True
 
             sage: x = polygen(ZZ, 'x')
             sage: F.<a> = NumberField(x^2 - x - 1)
-            sage: B.<i,j,k> = QuaternionAlgebra(F, 2*a, F(-1))
-            sage: B.discriminant()
-            Fractional ideal (2)
+            sage: QuaternionAlgebra(F, 2*a, F(-1)).is_totally_definite()
+            False
 
-            sage: QuaternionAlgebra(QQ[sqrt(2)], 3, 19).discriminant()                  # needs sage.symbolic
-            Fractional ideal (1)
+            sage: QuaternionAlgebra(RR(2.),1).is_totally_definite()
+            Traceback (most recent call last):
+            ...
+            ValueError: base field must be rational numbers or a number field
         """
-        if not is_RationalField(self.base_ring()):
-            try:
-                F = self.base_ring()
-                return F.hilbert_conductor(self._a, self._b)
-            except NotImplementedError:
-                raise ValueError("base field must be rational numbers or number field")
-        else:
-            return ZZ.prod(self.ramified_primes())
+        F = self.base_ring()
+        if is_RationalField(F):
+            return self.is_definite()
+
+        if F not in NumberFields():
+            raise ValueError("base field must be rational numbers or a number field")
+
+        E = F.real_embeddings()
+        return [e for e in E if F.hilbert_symbol(self._a, self._b, e) == -1] == E
+
+    @cached_method
+    def ramified_places(self, inf=True):
+        r"""
+        Return the places of the base number field at which this
+        quaternion algebra ramifies.
+
+        Note: The initial choice of primes (for the base field `\QQ`)
+        respectively of prime ideals (in the number field case) to check
+        ramification for is motivated by 12.4.12(a) in [Voi2021]_. The
+        restriction to real embeddings is due to 14.5.8 in [Voi2021]_.
+
+        INPUT:
+
+        - ``inf`` -- (default: ``True``)
+
+        OUTPUT:
+
+        The non-Archimedean (AKA finite) places at which the quaternion
+        algebra ramifies (given as elements of `\ZZ`, sorted small to
+        large, if the algebra is defined over the rational field `\QQ`,
+        respectively as fractional ideals of the number field's ring of
+        integers, otherwise) and, if ``inf`` is set to ``True``, also the
+        Archimedean (AKA infinite) places at which the quaternion algebra
+        ramifies (given by real embeddings of the base field).
+
+        EXAMPLES::
+
+            sage: QuaternionAlgebra(210,-22).ramified_places()
+            ([2, 3, 5, 7], [])
+
+            sage: QuaternionAlgebra(-1, -1).ramified_places()
+            ([2],
+             [Ring morphism:
+                From: Rational Field
+                To:   Real Field with 53 bits of precision
+                Defn: 1 |--> 1.00000000000000])
```

Comment:
```suggestion
                Defn: 1 |--> 1.00000000000000])
                
            ::
```
You can use `::` to separate these blocks of examples. This is what it looks like right now (it should be in separate blocks)

![Screenshot 2024-03-27 at 05 19 40](https://github.com/sagemath/sage/assets/83517584/0fb9b47e-096d-4a20-a42e-59fc733304ff)